packages/core/src/domain/session/sessionStoreOperations.spec.ts

```diff
@@ -0,0 +1,248 @@
+import { stubCookie, mockClock } from '../../../test'
+import { isChromium } from '../../tools/utils/browserDetection'
+import { SESSION_EXPIRATION_DELAY } from './sessionConstants'
+import { initCookieStore, SESSION_COOKIE_NAME, toSessionString } from './sessionCookieStore'
+import type { SessionState, SessionStore } from './sessionStore'
+import {
+  processSessionStoreOperations,
+  isLockEnabled,
+  LOCK_MAX_TRIES,
+  LOCK_RETRY_DELAY,
+} from './sessionStoreOperations'
+
+describe('process operations mechanism', () => {
+  const COOKIE_OPTIONS = {}
+  let initialSession: SessionState
+  let otherSession: SessionState
+  let processSpy: jasmine.Spy<jasmine.Func>
+  let afterSpy: jasmine.Spy<jasmine.Func>
+  let cookie: ReturnType<typeof stubCookie>
+  let cookieStorage: SessionStore
+
+  beforeEach(() => {
+    cookieStorage = initCookieStore(COOKIE_OPTIONS)
+    initialSession = { id: '123', created: '0' }
+    otherSession = { id: '456', created: '100' }
+    processSpy = jasmine.createSpy('process')
+    afterSpy = jasmine.createSpy('after')
+    cookie = stubCookie()
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏instead of stubbing cookie and providing a cookie store, could we we provide a store stub directly?

Reply:
👍 done in https://github.com/DataDog/browser-sdk/pull/2261/commits/8142e0c782776647b4579852a6f49aee2466c529